flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/operator/common/dataproc/vector/VectorToColumnsMapper.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.operator.common.dataproc.vector;
+
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.ml.api.misc.param.Params;
+import org.apache.flink.ml.common.linalg.DenseVector;
+import org.apache.flink.ml.common.linalg.SparseVector;
+import org.apache.flink.ml.common.linalg.Vector;
+import org.apache.flink.ml.common.mapper.Mapper;
+import org.apache.flink.ml.common.utils.OutputColsHelper;
+import org.apache.flink.ml.common.utils.TableUtil;
+import org.apache.flink.ml.params.dataproc.vector.VectorToColumnsParams;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Preconditions;
+
+import java.util.Arrays;
+
+/**
+ * This mapper maps vector to table columns, and the table is created with the first
+ * colSize value of the vector.
+ * For sparse vector without given size, it will be treated as vector with infinite size.
+ * If the colSize is larger than the vector size, we'll throw exception;
+ * If the colSize is not larger than the vector size, we'll select the first
+ * colSize value of the vector.
+ */
+public class VectorToColumnsMapper extends Mapper {
+	private int colSize;
+	private int idx;
+	private OutputColsHelper outputColsHelper;
+
+	public VectorToColumnsMapper(TableSchema dataSchema, Params params) {
+		super(dataSchema, params);
+		String selectedColName = this.params.get(VectorToColumnsParams.SELECTED_COL);
+		idx = TableUtil.findColIndex(dataSchema.getFieldNames(), selectedColName);
+		Preconditions.checkArgument(idx >= 0, "Can not find column: " + selectedColName);
+		String[] outputColNames = this.params.get(VectorToColumnsParams.OUTPUT_COLS);
+		Preconditions.checkArgument(null != outputColNames,
+			"VectorToTable: outputColNames must set.");
+		this.colSize = outputColNames.length;
+		TypeInformation[] types = new TypeInformation[colSize];
+		Arrays.fill(types, Types.DOUBLE);
+		this.outputColsHelper = new OutputColsHelper(dataSchema, outputColNames, types,
+			this.params.get(VectorToColumnsParams.RESERVED_COLS));
```

Comment:
Let's refactor this part out as a private static function: `constructOutputColsHelper` ? I can see @ex00 's concerns that the constructor is way too complex. In principle the constructor should be simpler and easy to understand. One example is to have this as:
```
public VectorToColumnsMapper(TableSchema dataSchema, Params params) {
  this(dataSchema, params, constructOutputColsHelper(dataSchema, params));
}

public VectorToColumnsMapper(TableSchema dataSchema, Params params, OutputColsHelper outputColsHelper) {
  super(dataSchema, params);
  this.outputColsHelper = outputColsHelper;
}
```